practice-two/src/components/ProductDetails.jsx

```diff
@@ -42,6 +44,10 @@ const ProductDetails = ({
     onToggleProductFavorite(product.id);
   };
 
+  const handleAddToCart = () => {
+    onAddToCart(product.id);
```

Comment:
product is [] so why we have `product.id` 
![image](https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/27277174/edb25ffd-6c17-46e3-8cf1-d581f2a547ae)

Reply:
There seems to be an inconsistency in the code. The product prop is given a default value of an empty array ( product = []) in the function parameter, but in the code it is treated as an object with properties like product.id, which would lead to error. Changed default value for products